infra/feast-operator/internal/controller/services/services.go

```diff
@@ -917,3 +924,13 @@ func getProbeHandler(feastType FeastServiceType, tls *feastdevv1alpha1.TlsConfig
 		},
 	}
 }
+
+func IsDeploymentAvailable(conditions []appsv1.DeploymentCondition) bool {
+	for _, condition := range conditions {
+		if condition.Type == appsv1.DeploymentAvailable {
+			return condition.Status == corev1.ConditionTrue
+		}
+	}
+
+	return false
```

Comment:
does something like this do the same?
```suggestion
        return apimeta.IsStatusConditionTrue(conditions, appsv1.DeploymentAvailable)
```

Reply:
if my suggestion does the job, we may not need the new `IsDeploymentAvailable`... this check could be done directly.

Reply:
You can't do that since `deployment.Status.Conditions` is of type `DeploymentCondition` and not `metav1.Condition` thats the reason I didn't do it in the first place

Reply:
> if my suggestion does the job, we may not need the new `IsDeploymentAvailable` method... this check could be done directly.

You can't do that since deployment.Status.Conditions is of type DeploymentCondition and not metav1.Condition thats the reason I didn't do it in the first place

Reply:
ok, makes sense... thanks